cpp/src/barretenberg/benchmark/honk_bench/benchmark_utilities.hpp

```diff
@@ -0,0 +1,224 @@
+ */
+template <typename Composer> void generate_basic_arithmetic_circuit(Composer& composer, size_t num_gates)
+{
+    plonk::stdlib::field_t a(plonk::stdlib::witness_t(&composer, barretenberg::fr::random_element()));
+    plonk::stdlib::field_t b(plonk::stdlib::witness_t(&composer, barretenberg::fr::random_element()));
+    plonk::stdlib::field_t c(&composer);
+    for (size_t i = 0; i < (num_gates / 4) - 4; ++i) {
+        c = a + b;
+        c = a * c;
+        a = b * b;
+        b = c * c;
+    }
+}
+
+/**
+ * @brief Generate test circuit with specified number of sha256 hashes
+ *
+ * @param composer
+ * @param num_iterations
+ */
+template <typename Composer> void generate_sha256_test_circuit(Composer& composer, size_t num_iterations)
+{
+    std::string in;
+    in.resize(32);
+    for (size_t i = 0; i < 32; ++i) {
+        in[i] = 0;
+    }
+    proof_system::plonk::stdlib::packed_byte_array<Composer> input(&composer, in);
+    for (size_t i = 0; i < num_iterations; i++) {
+        input = proof_system::plonk::stdlib::sha256<Composer>(input);
+    }
+}
+
+/**
+ * @brief Generate test circuit with specified number of keccak hashes
+ *
+ * @param composer
+ * @param num_iterations
+ */
+template <typename Composer> void generate_keccak_test_circuit(Composer& composer, size_t num_iterations)
+{
+    std::string in = "abcdefghijklmnopqrstuvwxyz0123456789abcdefghijklmnopqrstuvwxyz01";
+
+    proof_system::plonk::stdlib::byte_array<Composer> input(&composer, in);
+    for (size_t i = 0; i < num_iterations; i++) {
+        input = proof_system::plonk::stdlib::keccak<Composer>::hash(input);
+    }
+}
+
+/**
+ * @brief Generate test circuit with specified number of ecdsa verifications
+ *
+ * @param composer
+ * @param num_iterations
+ */
+template <typename Composer> void generate_ecdsa_verification_test_circuit(Composer& composer, size_t num_iterations)
+{
+    using curve = proof_system::plonk::stdlib::secp256k1<Composer>;
+    using fr = typename curve::fr;
+    using fq = typename curve::fq;
+    using g1 = typename curve::g1;
+
+    std::string message_string = "Instructions unclear, ask again later.";
+
+    crypto::ecdsa::key_pair<fr, g1> account;
+    account.private_key = curve::fr::random_element();
+    account.public_key = curve::g1::one * account.private_key;
+
+    crypto::ecdsa::signature signature =
+        crypto::ecdsa::construct_signature<Sha256Hasher, fq, fr, g1>(message_string, account);
+
+    bool first_result =
+        crypto::ecdsa::verify_signature<Sha256Hasher, fq, fr, g1>(message_string, account.public_key, signature);
+
+    std::vector<uint8_t> rr(signature.r.begin(), signature.r.end());
+    std::vector<uint8_t> ss(signature.s.begin(), signature.s.end());
+    uint8_t vv = signature.v;
+
+    typename curve::g1_bigfr_ct public_key = curve::g1_bigfr_ct::from_witness(&composer, account.public_key);
+
+    proof_system::plonk::stdlib::ecdsa::signature<Composer> sig{ typename curve::byte_array_ct(&composer, rr),
```

Comment:
I'd generate several separate  unique signatures, because at some point this can devolve into basically one, if we do optimisations correctly.

Reply:
good call, done